nix/haskell.nix

```diff
@@ -13,33 +13,44 @@
 }:
 let
 
+  src = haskell-nix.haskellLib.cleanGit {
+      name = "cardano-prelude-src";
+      src = ../.;
+  };
+
   # This creates the Haskell package set.
   # https://input-output-hk.github.io/haskell.nix/user-guide/projects/
-  pkgSet = haskell-nix.cabalProject {
-      src = haskell-nix.haskellLib.cleanGit { src = ../.; };
-      ghc = buildPackages.haskell-nix.compiler.${compiler};
+  pkgSet = haskell-nix.cabalProject  (lib.optionalAttrs stdenv.hostPlatform.isWindows {
+    # FIXME: without this deprecated attribute, db-converter fails to compile directory with:
+    # Encountered missing dependencies: unix >=2.5.1 && <2.9
+    ghc = buildPackages.haskell-nix.compiler.${compiler};
```

Comment:
We should be able to remove this now.  The `compiler-nix-name = compiler;` on its own should be enough.